src/main/resources/META-INF/services/bisq.core.payment.validation.SpecificAltCoinAddressValidator

```diff
@@ -0,0 +1,2 @@
+bisq.core.payment.validation.altcoins.bch.BchAddressValidator
+bisq.core.payment.validation.altcoins.ella.EllaAddressValidator
```

Comment:
What I had in mind here for the SPI is something that would be called, e.g. a `TokenListingProvider`, and implementations of that SPI would provide everything necessary to list a token, including any custom validators like this. So think self-contained, higher-level.

Reply:
As I mentioned in Slack just after the comment above, perhaps what we want for naming is just `TokenProvider`.

And actually, we may want to take this opportunity to raise the level of abstraction one degree and start referring to these things in general as "assets", as-in "crypto assets". That term has emerged pretty strongly as the most general class that includes everything from BTC itself to altcoins (e.g. ETH, LTC) to tokens, e.g. all things ERC-20 on top of ETH.

We already distinguish within Bisq between altcoins (coins with their own chain) and "tokens" (coins that ride on another chain, e.g. ERC-20). Asset is the missing generalization here.

So I could see us calling the SPI here `AssetProvider`, and updating our documentation to talk about "listing an asset" on Bisq, instead of "listing a token".

So the `AssetProvider` would be one-stop shopping for all your Bisq asset listing needs: it would define your asset name, asset ticker symbol, asset address validation, etc.

Reply:
We use isAsset in the CryptoCurrency to distinguish between pure altcoins (own blockchain) and assets like ERC-20. So far it is only used to show an extra popup.